spec/suites/map/handler/Map.TapHoldSpec.js

```diff
@@ -0,0 +1,209 @@
+			zoom: 13,
+			tapHold: true
+		});
+		map.on('contextmenu', spy);
+
+		posStart.target = el;
+		posNear.target = el;
+		posFar.target = el;
+	});
+
+	afterEach(function () {
+		happen.once(el, {type: 'touchend'});
+		for (var id = 0; id <= 2; id++) { // reset pointers (for prosphetic-hand)
+			happen.once(el, {type: 'pointercancel', pointerId:id});
+		}
+		clock.restore();
+		map.remove();
+		document.body.removeChild(el);
+	});
+
+	it('fires synthetic contextmenu after hold delay>600', function () {
+		happen.once(el, {type: 'touchstart', touches: [posStart]});
+		happen.once(el, L.extend({type: 'pointerdown', pointerId:0}, posStart));
+		clock.tick(550);
+
+		expect(spy.notCalled).to.be.ok();
+
+		clock.tick(100);
+
+		expect(spy.called).to.be.ok();
+		expect(spy.calledOnce).to.be.ok();
+
+		var event = spy.lastCall.args[0];
+		expect(event.type).to.be('contextmenu');
+		expect(event.originalEvent._simulated).to.be.ok();
+	});
+
+	it('does not fire contextmenu when touches > 1', function () {
+		happen.once(el, {type: 'touchstart', touches: [posStart]});
+		happen.once(el, L.extend({type: 'pointerdown', pointerId:0}, posStart));
+		clock.tick(100);
+		happen.once(el, {type: 'touchstart', touches: [posStart, posNear]});
+		happen.once(el, L.extend({type: 'pointerdown', pointerId:1}, posNear));
+		clock.tick(550);
+
+		expect(spy.notCalled).to.be.ok();
+	});
+
+	it('does not fire contextmenu when touches > 1 (case:2)', function () {
+		happen.once(el, {type: 'touchstart', touches: [posStart]});
+		happen.once(el, L.extend({type: 'pointerdown', pointerId:0}, posStart));
+		clock.tick(100);
+		happen.once(el, {type: 'touchstart', touches: [posStart, posNear]});
+		happen.once(el, L.extend({type: 'pointerdown', pointerId:1}, posNear));
+		clock.tick(100);
+		happen.once(el, {type: 'touchend', touches: [posStart]});
+		happen.once(el, L.extend({type: 'pointerup', pointerId:0}, posNear));
+		clock.tick(450);
+
+		expect(spy.notCalled).to.be.ok();
+	});
+
+	(L.Browser.pointer ? it : it.skip)('ignores events from mouse', function () {
+		happen.once(el, L.extend({type: 'pointerdown', pointerId:0, pointerType:'mouse'}, posStart));
+		clock.tick(650);
+
+		expect(spy.notCalled).to.be.ok();
+	});
+
+	it('does not conflict with native contextmenu', function () {
+		happen.once(el, {type: 'touchstart', touches: [posStart]});
+		happen.once(el, L.extend({type: 'pointerdown', pointerId:0}, posStart));
+		clock.tick(550);
+
+		happen.once(el, {type: 'contextmenu'});
+
+		clock.tick(100);
+
+		expect(spy.called).to.be.ok();
+		expect(spy.calledOnce).to.be.ok();
+		expect(spy.lastCall.args[0].originalEvent._simulated).not.to.be.ok();
+
+		// Note: depending on tapHoldDelay value it's also possible that native contextmenu may come after simulated one
+		//       and the only way to handle this gracefully - increase tapHoldDelay value.
+		//       Anyway that is edge case, as tapHold is meant for browsers where native contextmenu is not fired on touch.
```

Comment:
Should we add this somewhere in the code or in the docs?

Reply:
It is about implementation details, and not useful in docs.